crates/adapters/src/util.rs

```diff
@@ -60,3 +64,198 @@ macro_rules! dyn_event {
         }
     };
 }
+
+/// A job queue that dispatches work to a pool of tokio tasks.
```

Comment:
could be the standard framework for all adapters?

Reply:
That's kind of the plan. Problem is, this requires async, so can only be easily combined with tokio-based connectors. We could have a sync version of this, but that can potentially be more expensive, especially for short messages.